### ISSUE TITLE
[HOLD for payment 2023-02-24] [$2000] Chat -If you mark a message as unread, then delete it -"New messages" button does not dissapear

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#14649]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com
2. Log i with any account
3. Navigate to a 1:1 conversation that has a large history of messages
4. Press and hold/hover over your message and mark the message as "Unread"
5. Delete marked message
6. Scroll the conversation a little

## Expected Result:
The Green line should not be shown after deleting a marked message, also the "New messages" button at the top of the page when scrolling.

## Actual Result:
If you mark a message as unread and then delete it, a "New messages" button does not disappear. Or if you delete messages not at the end but in the middle of the chat, the green line does not disappear.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1,2,61,0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/215277014-6b7b3d41-5760-4602-902d-960c4d7b3409.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01418691c14f3167db</li>
        <li>Upwork Job ID: 1621665339139534848</li>
        <li>Last Price Increase: 2023-02-11</li>
    </ul>
</details>

**Comment by mateocole** [comment]
@NikkiWines is this something we will want to work on fixing internally?

**Comment by NikkiWines** [comment]
Hmm, yeah I'm not sure what the intended behavior is here to be honest. Not sure if we'd consider this part of the same issue (or an issue at all) but the line and banner seem to persist even if you don't delete the message and scroll past the message marked as unread. cc: @roryabraham tagging you cause slack history shows you've mentioned the unread indicator a couple of times

https://user-images.githubusercontent.com/16747822/216163054-f0ac0690-addb-4cba-a3df-a109d4adea16.mov

**Comment by roryabraham** [comment]
Trying my best to clarify the expected behavior here:

- If you mark a comment as unread, then generally the only thing that should make that comment unread or cause the new message indicator to disappear is if you leave the report and come back.
- If you mark a comment as unread, then delete it, the expected behavior _depends_:
    - If the comment you marked as unread then deleted is the last reportAction in a report, then the report should go back to the _read_ state, and the new message indicator should disappear
    - If the comment you marked as unread then deleted had reportActions after it in the report, then the report should remain in the _unread_ state, and the new message indicator should appear on the next-oldest reportAction.

Does that make sense/

**Comment by roryabraham** [comment]
So @NikkiWines what you've shown in [this screenshot](https://github.com/Expensify/App/issues/14649#issuecomment-1412725554) is expected. 👍🏼

**Comment by roryabraham** [comment]
Looking closer at [the video in the OP](https://github.com/Expensify/App/issues/14649#issue-1560916454), it looks like this is happening as expected:

1. Mark an older comment as unread
1. Delete that comment

*Expected behavior:* The report remains unread and the new message indicator appears on the next-oldest reportAction ✅ 

However, this is _not_ happening as expected:

### Steps to reproduce:

1. Send a message in a chat
1. Mark it as unread
1. Delete it.

### Expected behavior
- The report should go back to the _read_ state
- The new message indicator should disappear
- If you scroll up, then the "new message" pill should not be present.

### Actual behavior
- The report goes back to the _read_ state ✅ 
- The new message indicator disappears ✅ 
- If you scroll up, then the "new message" pill is visible ❌

**Comment by NikkiWines** [comment]
Awesome, thanks for clarifying @roryabraham 🙌  From what you've written it sounds like there is still something to fix here and that the issue can be `External`.

**Comment by tienifr** [comment]
# Proposal 

This is because we're missing a case when deciding whether to update the `newMarkerReportActionID`

```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..87a81ab1a0 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -207,7 +207,7 @@ class ReportActionsView extends React.Component {
 
         // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
         // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
+        if (ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
             this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
         }
 

```

**Comment by tienifr** [comment]
# Proposal 2

We can add the condition to only run the update when there's a current `newMarkerReportActionID`, to minimize the time the code run

```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..87a81ab1a0 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -207,7 +207,7 @@ class ReportActionsView extends React.Component {
 
         // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
         // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
+        if (this.state.newMarkerReportActionID && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
             this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
         }
 

```

**Comment by anupp-11** [comment]
PROPOSAL

If the report is marked as unread and then deleted, it would still show as having a new message even though the report is no longer present. To fix this issue, we could add a check to see if the report still exists before trying to access its properties and actions. If the report has been deleted, you could reset the state of the component.

**Comment by tienifr** [comment]
Both [proposal 1](https://github.com/Expensify/App/issues/14649#issuecomment-1416651174) and [proposal 2](https://github.com/Expensify/App/issues/14649#issuecomment-1416657851) working well after the fix:


https://user-images.githubusercontent.com/113963320/216748831-e3933685-84ba-45a4-96dc-a2e35cf38e17.mp4

**Comment by tienifr** [comment]
# Proposal 3

We can furthermore strengthen the setState condition, to only update the `newMarkerReportActionID` state only when it equals to the deleted action

```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..cedcc6d84f 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -207,8 +207,15 @@ class ReportActionsView extends React.Component {
 
         // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
         // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
+        let filterReportActionsForDisplay = ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions);
+        if (this.state.newMarkerReportActionID && filterReportActionsForDisplay.length > this.sortedAndFilteredReportActions.length) {
+            let deletedReportActionIDs = _.difference(
+                filterReportActionsForDisplay.map(item => item.reportActionID),
+                this.sortedAndFilteredReportActions.map(item => item.reportActionID)
+            )
+            if (deletedReportActionIDs.includes(this.state.newMarkerReportActionID)) {
+                this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
+            }
         }
 
         // When the user navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
```

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..d114774303 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -211,11 +211,9 @@ class ReportActionsView extends React.Component {
             this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
         }
 
-        // When the user navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
-        // The next time we navigate to the same report (e.g. by swiping or tapping the LHN row) we want the new marker to clear.
-        const didSidebarOpen = !prevProps.isDrawerOpen && this.props.isDrawerOpen;
-        const didUserNavigateToSidebarAfterReadingReport = didSidebarOpen && !ReportUtils.isUnread(this.props.report);
-        if (didUserNavigateToSidebarAfterReadingReport) {
+        // If the report is read and we still have the new marker set, then clear the marker.
+        // (e.g. When the user navigates to the LHN or deletes the last message that is also the only unread message)
+        if (this.state.newMarkerReportActionID && !ReportUtils.isUnread(this.props.report)) {
             this.setState({newMarkerReportActionID: ''});
         }
 
```

### RCA
If the user deletes the last message that is also the only unread message, the report is marked as read but we don't clear the `newMarkerReportActionID`. We had a similar case with LHN, where navigating to the LHN did not clear the new marker either.

The fix is to add a condition that checks if the report is read then make sure that the new marker is not set. if it's set then clear it.
This will cover all the possible cases including the one mentioned in this issue and the LHN one (I have removed the explicit LHN fix since it's redundant now).

**Comment by tienifr** [comment]
# Proposal 4

I think `newMarkerReportActionID` should be something that's derived from the state (the report). Currently what we do is, we're not syncing the `newMarkerReportActionID` to the actual state of the report but we try to think of cases where the `newMarkerReportActionID` might change and then add handling logic for to update it. I think we should always calculate the `newMarkerReportActionID` based on the state of the report inside `componentDidUpdate` to avoid complicated logic and edge cases that cause bugs. Please note this is not less performant than the current approach since the calculation is a O(n) operation based on the number of reportActions, and the current approach (for checking the cases) is also O(n) (the `ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length) calculation`.

```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..01c14f484f 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -197,35 +197,10 @@ class ReportActionsView extends React.Component {
         const didScreenSizeIncrease = prevProps.isSmallScreenWidth && !this.props.isSmallScreenWidth;
         const didReportBecomeVisible = isReportFullyVisible && (didSidebarClose || didScreenSizeIncrease);
         if (didReportBecomeVisible) {
-            this.setState({
-                newMarkerReportActionID: ReportUtils.isUnread(this.props.report)
-                    ? ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)
-                    : '',
-            });
             this.openReportIfNecessary();
         }
 
-        // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
-        // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
-        }
-
-        // When the user navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
-        // The next time we navigate to the same report (e.g. by swiping or tapping the LHN row) we want the new marker to clear.
-        const didSidebarOpen = !prevProps.isDrawerOpen && this.props.isDrawerOpen;
-        const didUserNavigateToSidebarAfterReadingReport = didSidebarOpen && !ReportUtils.isUnread(this.props.report);
-        if (didUserNavigateToSidebarAfterReadingReport) {
-            this.setState({newMarkerReportActionID: ''});
-        }
-
-        // Checks to see if a report comment has been manually "marked as unread". All other times when the lastReadTime
-        // changes it will be because we marked the entire report as read.
-        const didManuallyMarkReportAsUnread = (prevProps.report.lastReadTime !== this.props.report.lastReadTime)
-            && ReportUtils.isUnread(this.props.report);
-        if (didManuallyMarkReportAsUnread) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
-        }
+        this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
 
         // Ensures subscription event succeeds when the report/workspace room is created optimistically.
         // Check if the optimistic `OpenReport` or `AddWorkspaceRoom` has succeeded by confirming

```

If we do the above, there could be other places where the update of `newMarkerReportActionID` can be removed. I'm happy to add those if there's interest to go with this approach.

**Comment by s77rt** [comment]
@thesahindia Can you share a video on what you mean exactly? I think if the issue you are referring is too different (have another root cause) then it should be handled separately (should we ask on slack?)

**Comment by s77rt** [comment]
@thesahindia Thanks! I have investigated the mentioned bug and I confirm that it's completely unrelated to this issue.
The bug is that the `lastReadTime` is not getting updated thus the report is still marked as unread.

Here is the (supposed) flow:
1. If you open a report `isDrawerOpen` will be set to `false`
2. Swipe back to LHN `isDrawerOpen` will be set to `true`
3. Open report again `isDrawerOpen` will be set to `false`
4. In `componentDidUpdate` we call `openReportIfNecessary` if `didReportBecomeVisible` condition is met (which is met if you we close the sidebar just as the above steps says)

So, the logic is correct and the bug should not exist except that step 2 will not occur because we are using `<Freeze />` that freezes everything and messes with the states, It's very react (cycle) unfriendly. The good news is that we are moving away from Drawer into Stack and with that `<Freeze />` will no longer make sense and it will be removed, and the bug will be fixed.

**Comment by s77rt** [comment]
You can confirm that the bug is caused by `<Freeze />` by disabling it.
```diff
diff --git a/src/pages/home/ReportScreen.js b/src/pages/home/ReportScreen.js
index 1be75b3e1d..e8b48ce17f 100644
--- a/src/pages/home/ReportScreen.js
+++ b/src/pages/home/ReportScreen.js
@@ -222,7 +222,7 @@ class ReportScreen extends React.Component {
                 style={screenWrapperStyle}
             >
                 <Freeze
-                    freeze={freeze}
+                    freeze={false}
                     placeholder={(
                         <>
                             <ReportHeaderSkeletonView animate={animatePlaceholder} />
```

**Comment by roryabraham** [comment]
> You can confirm that the bug is caused by <Freeze /> by disabling it.

@neil-marcellini @thesahindia if this is indeed true I'd recommend we put this issue on HOLD for the react-navigation reboot since we'll be getting rid of `<Freeze />` as part of that project anyways, rather than building any additional logic to workaround the limitations of `<Freeze />`

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by s77rt** [comment]
@roryabraham I was referring to the other reported bug by @thesahindia [here](https://github.com/Expensify/App/issues/14649#issuecomment-1419565503). The original reported bug in OP still remains and it's unrelated to `<Freeze />`.

**Comment by tienifr** [comment]
thanks @thesahindia for the feedback.

If I understand correctly, it might be an existing behavior. Here's what I captured from latest `main`, we also don't see `the new message line indicator` when navigating to chat B:

https://user-images.githubusercontent.com/113963320/217237192-3fbc3f08-1531-4af8-9cb8-b27ae6eb2504.mov

**Comment by s77rt** [comment]
### Proposal (Updated)
```diff
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997a..384a08f696 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -211,11 +211,9 @@ class ReportActionsView extends React.Component {
             this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
         }
 
-        // When the user navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
-        // The next time we navigate to the same report (e.g. by swiping or tapping the LHN row) we want the new marker to clear.
-        const didSidebarOpen = !prevProps.isDrawerOpen && this.props.isDrawerOpen;
-        const didUserNavigateToSidebarAfterReadingReport = didSidebarOpen && !ReportUtils.isUnread(this.props.report);
-        if (didUserNavigateToSidebarAfterReadingReport) {
+        // If the report is read and we still have the new marker set, then clear the marker.
+        // (e.g. When the user navigates to the LHN or deletes the last message that is also the only unread message)
+        if (this.state.newMarkerReportActionID && !ReportUtils.isUnread(prevProps.report) && !ReportUtils.isUnread(this.props.report)) {
             this.setState({newMarkerReportActionID: ''});
         }
 
```

### Updated
Fixed the above [regression](https://github.com/Expensify/App/issues/14649#issuecomment-1420628735)

**Comment by s77rt** [comment]
@thesahindia Can you please share a video for that? As far as I tested I still get the new marker

https://user-images.githubusercontent.com/16493223/217325985-dca08093-2674-43d2-b0d1-bdbc3e67247e.mp4

**Comment by s77rt** [comment]
I think that's the expected behavior. When you open the chat you make a request to the server and update the lastReadTime making the report unread. You can see that the bold styling in LHN will also be removed. If anything I'm adding more consistency to the app (bolded LHN <=> New marker). Thoughts?

**Comment by tienifr** [comment]
@thesahindia I shared a video in the [last comment](https://github.com/Expensify/App/issues/14649#issuecomment-1420646882). That was the code from main.

What I did was:
1. Login as user 1, Open chat A
2. Use another device to send message to chat B of user 1
3. Navigate to chat B as user 1, observe that there's no `New message` marker

Here's another video from current staging:

https://user-images.githubusercontent.com/113963320/217337007-f17bc920-fb2d-4b47-a282-b860f110df09.mov

Not sure if that's the steps to reproduce that you mean.

**Comment by s77rt** [comment]
@thesahindia @tienifr Yes, I confirm the regression

@thesahindia Sure, let's ask on slack.

**Comment by s77rt** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
If you send a message, mark it as unread then deleted it. You would still see a meaningless new marker indicator with "New messages" floating button appearing.

### What is the root cause of that problem?
After we set a message as unread we update the [newMarkerReportActionID](https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/home/report/ReportActionsView.js#L226-L228). Now `newMarkerReportActionID` holds a value X where X is that message ID (termed report action id). If we delete that report action, the `newMarkerReportActionID` will still hold the same value X referring to a no longer existing report action (message).

### What changes do you think we should make in order to solve the problem?
Ensure that the `newMarkerReportActionID` is referring a report action that actually exists. Otherwise recalculate a new marker position. This will most likely be implemented [here](https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/home/report/ReportActionsView.js#L213) and as pseudo-code that would be:
```
if newMarkerReportActionID is set
    Verify that the report action that has ID=newMarkerReportActionID exists
    If it does not exist
        newMarkerReportActionID = recalculateNewMarkerReportActionID()
    endif
endif
```

### What alternative solutions did you explore? (Optional)
Other than the stated in the previous comments, none.

**Comment by abdulrahuman5196** [comment]
@thesahindia  I have made a proposal here - https://github.com/Expensify/App/issues/15019#issuecomment-1425434440 which would also solve this issue.

**Comment by MelvinBot** [comment]
@neil-marcellini @mateocole @thesahindia this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by neil-marcellini** [comment]
@thesahindia I don't think we should hold this issue on [the proposal](https://github.com/Expensify/App/issues/15019#issuecomment-1425434440) from #15019. I don't think that proposal has clearly stated the root cause and there's no consensus in Slack yet about whether we want to move forward with that approach.

If we like @s77rt's proposal then I think we should move forward with it. If we find a better implementation later, it can be changed, but I think we should focus on fixing this bug quickly. I'll give his proposal a review now myself.

**Comment by neil-marcellini** [comment]
@thesahindia I like @s77rt's [proposal](https://github.com/Expensify/App/issues/14649#issuecomment-1423945717) and I think we should move forward with it.

**Comment by abdulrahuman5196** [comment]
@neil-marcellini  I have updated the proposal with clear information as per slack request - https://github.com/Expensify/App/issues/15019#issuecomment-1425434440

**Comment by MelvinBot** [comment]
📣 @s77rt You have been assigned to this job by @neil-marcellini!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by s77rt** [comment]
Applied in [Upwork](https://www.upwork.com/jobs/~01418691c14f3167db).
[PR](https://github.com/Expensify/App/pull/15144) is ready.

**Comment by neil-marcellini** [comment]
Merged!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.73-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15144


If no regressions arise, payment will be issued on 2023-02-24. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@thesahindia / @neil-marcellini] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@thesahindia / @neil-marcellini] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@thesahindia / @neil-marcellini] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@s77rt] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mateocole] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
**Regression Test Proposal**

1. Login to any account
2. Go to any chat
3. Send a message
4. Mark that message as unread
5. Delete that message
6. Scroll up
7. Verify that you don't see the "New messages" button at the top

**Comment by neil-marcellini** [comment]
Regarding the first 3 checklist items, I think this bug may have been expected after the project to deprecate sequence numbers, so I don't think it's worth tracking down the PR and commenting there. I also don't think we need to raise a discussion because we expected some small bugs after such a large refactor.

@s77rt I think the regression test looks good. @mateocole would you please create an issue to add the regression test and handle payment?

**Comment by neil-marcellini** [comment]
@mateocole bump

**Comment by neil-marcellini** [comment]
@mateocole bump again

**Comment by neil-marcellini** [comment]
Bumped again [in Slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1678311046747969). I also asked if someone can take over.

**Comment by mateocole** [comment]
Alright offers sent!

**Comment by mateocole** [comment]
Okay, both payments have been sent

**Comment by mateocole** [comment]
50% Bonus sent
